### PR TITLE
Return list of clients in format FreeRadius expects

### PR DESCRIPTION
**Pull request by RyanMacG**
This allows us to return the list of allowed clients from the database
in a format that FreeRadius will accept.

We also no longer want to use the read replica for retrieving this data.

### CODE REVIEW

File: app.rb [hunk 1]
```diff
@@ -31,4 +35,17 @@ def connect_to_db
   def valid_key?
     params.dig(:key) == ENV.fetch('FRONTEND_API_KEY')
   end
+
+  def convert_for_freeradius(sites_and_ips)
+    sites = []
+    sites_and_ips.each do |site|
```
Comment (emileswarts) [inline app.rb]: .map ?

File: app.rb [hunk 2]
```diff
@@ -7,10 +7,14 @@ class App < Sinatra::Base
   end
 
   get '/clients' do
+    return '' unless ENV.fetch('FRONTEND_API_KEY')
```
Comment (afoldesi-gds) [inline app.rb]: Move to valid_key? - should anything need it

File: app.rb [hunk 3]
```diff
@@ -7,10 +7,14 @@ class App < Sinatra::Base
...
+
     if valid_key?
-      db = connect_to_db    
-      tables = db.fetch('show tables')
-      tables.all.to_json
+      db = connect_to_db
+      sites_and_ips = db[:site].join(:siteip, site_id: :id)
```
Comment (afoldesi-gds) [inline app.rb]: Can you specify the fields here?

Comment (RyanMacG) [reply app.rb]: As far as I know not without passing raw SQL in and I don't see any real advantage to doing that.

File: app.rb [hunk 4]
```diff
@@ -7,10 +7,14 @@ class App < Sinatra::Base
...
 
   get '/clients' do
+    return '' unless ENV.fetch('FRONTEND_API_KEY')
+
     if valid_key?
-      db = connect_to_db    
-      tables = db.fetch('show tables')
-      tables.all.to_json
+      db = connect_to_db
+      sites_and_ips = db[:site].select(Sequel[:radkey], Sequel[:ip]).join(:siteip, site_id: :id)
```
Comment (afoldesi-gds) [inline app.rb]: Was this not just .select(:ip, :radkey) at the end of the join?

Comment (RyanMacG) [reply app.rb]: No, that didn't work and isn't mentioned in the Sequel documentation either

Comment (afoldesi-gds) [reply app.rb]: Well drat. .get() would return just one... IIRC

File: app.rb (hunk 2 above)
Comment (afoldesi-gds) [reply app.rb]: Emile had some changes for the valid key stuff just some time ago

File: app.rb [hunk 5]
```diff
@@ -1,34 +1,61 @@
...
+      host: ENV.fetch('DB_HOSTNAME'),
+      database: ENV.fetch('DB_NAME'),
+      user: ENV.fetch('DB_USER'),
+      password: ENV.fetch('DB_PASS')
     )
   end
 
-  def valid_key?
-    params.dig(:key) == ENV.fetch('FRONTEND_API_KEY')
+  def valid_api_request_key?
```
Comment (afoldesi-gds) [inline app.rb]: Oops.

**Pull request merged**